### ISSUE TITLE
BaseOptions override does not work for request timeout

### ISSUE THREAD

**Issue by coreyfreeman82** [nylo-core/nylo#104]
In the constructor of my api service I try to set the timeout of my requests to 10 seconds but whenever the request is sent it is still on 5 seconds.

ApiService:
```
class ApiService extends NyApiService {
  ApiService({BuildContext? buildContext})
      : super(buildContext, decoders: modelDecoders) {
    baseOptions = BaseOptions(
      connectTimeout: Duration(seconds: 10),
      receiveTimeout: Duration(seconds: 10),
      sendTimeout: Duration(seconds: 10),
    );
  }
}
```

Method to retrieve user:
```
Future<dynamic> checkUser() async {
    return await network(
      request: (request) {
        return request.get("/auth/me");
      },
      handleFailure: (DioException exception) {
        throw exception;
      },
    );
  }
```

Log shows 5 seconds instead of 10 and throws and exception because of it
```
I/flutter ( 3511): ╔╣ Request ║ POST 
I/flutter ( 3511): ║  https://my-site.com/api/v1/auth/me
I/flutter ( 3511): ╚══════════════════════════════════════════════════════════════════════════════════════════╝
I/flutter ( 3511): ╔ Headers 
I/flutter ( 3511): ╟ Content-type: application/json
I/flutter ( 3511): ╟ Accept: application/json
I/flutter ( 3511): ╟ contentType: application/json
I/flutter ( 3511): ╟ responseType: ResponseType.json
I/flutter ( 3511): ╟ followRedirects: true
I/flutter ( 3511): ╟ connectTimeout: 0:00:05.000000
I/flutter ( 3511): ╟ receiveTimeout: null
```

```
I/flutter ( 3511): ╔╣ DioError ║ DioExceptionType.connectionTimeout
I/flutter ( 3511): ║  The request connection took longer than 0:00:05.000000 and it was aborted. To get rid of this exception, try raising the RequestOptions.connectTimeout above the duration of 0:00:05.000000 or improve the response time of the server.
```

**Comment by agordn52** [comment]
Hi @coreyfreeman82,

I'll take a look into this now.
Thanks for reporting it 👍

**Comment by agordn52** [comment]
Hi @coreyfreeman82,

This is resolved in [v5.20.5](https://pub.dev/packages/nylo_framework)

You can now set the connectTimeout or any baseOptions from the constructor or by using the `network` helper.

**Using the constructor**
``` dart
ApiService({BuildContext? buildContext}) : super(
    buildContext, 
    decoders: modelDecoders, 
    baseOptions: (BaseOptions baseOptions) {
        return baseOptions
                  ..connectTimeout = Duration(seconds: 5)
                  ..sendTimeout = Duration(seconds: 5)
                  ..receiveTimeout = Duration(seconds: 5);
  }
```

**Using the `network` helper**

``` dart
Future fetchTestData() async {
    return await network(
        request: (request) => request.get("/endpoint-path"),
        // connectionTimeout: Duration(seconds: 5),
        // sendTimeout: Duration(seconds: 5),
        // receiveTimeout: Duration(seconds: 5),
    );
  }
```

Hope that helps 🔥
